errors_test.go

```diff
@@ -106,6 +106,54 @@ func TestCause(t *testing.T) {
 	}
 }
 
+func TestStack(t *testing.T) {
+	x := New("error")
+	y := WithStack(x).(*withStack).StackTrace()
+
+	tests := []struct {
+		err  error
+		want StackTrace
+	}{
+		{
+			err:  nil,
+			want: nil,
+		}, {
+			err:  (error)(nil),
+			want: nil,
+		}, {
+			err:  WithStack(io.EOF),
+			want: y,
+		}, {
+			err:  WithStack(WithStack(io.EOF)),
+			want: y,
+		}, {
+			err:  Wrap(WithStack(WithStack(io.EOF)), "test wrap"),
+			want: y,
+		}, {
+			err:  Cause(Wrap(WithStack(WithStack(io.EOF)), "test wrap")),
+			want: y,
+		}, {
+			err:  Cause(Cause(Wrap(WithStack(WithStack(io.EOF)), "test wrap"))),
+			want: y,
+		},
+	}
+
+	for i, tt := range tests {
+		got := Stack(tt.err)
+		if got == nil || tt.want == nil {
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("test %d: got %#v, want %#v", i+1, got, tt.want)
+			}
+
+			return
```

Comment:
```suggestion
			continue
```

Otherwise, we break out of the test after the first test, because definitively `tt.want == nil`.

Also, as written, by contract of the code design, we should expect all the other table tests to fail after this change.

Reply:
You are right. I will rewrite it.

Reply:
Thank you for your comments